synapse/http/matrixfederationclient.py

```diff
@@ -279,6 +279,7 @@ def __init__(self, hs, tls_client_options_factory):
             tls_client_options_factory,
             user_agent,
             hs.config.federation_ip_range_blacklist,
+            proxy_reactor=hs.get_reactor(),
```

Comment:
given that we already pass the ip blacklist into `MatrixFederationAgent`, why not move the construction of `BlacklistingReactorWrapper` down to it, rather than having to pass in two reactors?